core/src/main/java/org/apache/carbondata/core/scan/model/QueryModel.java

```diff
@@ -184,7 +188,9 @@ private static CarbonMeasure getCarbonMetadataMeasure(String name, List<CarbonMe
   }
 
   private static void setDimAndMsrColumnNode(FilterProcessVO processVO, ColumnExpression col,
-      boolean[] isFilterDimensions, boolean[] isFilterMeasures) {
+      boolean[] isFilterDimensions, boolean[] isFilterMeasures, CarbonTable table) {
+    boolean isDimensionFromExpression = col.isDimension();
+    boolean isMeasureFromExpression = col.isMeasure();
```

Comment:
How `ColumnExpression` knows about whether it is dimension or measure?

Reply:
`ColumnExpression` object have the flags whether it is dimension or measure, so im checking it

Reply:
I know `ColumnExpression` has flags, I am asking at this point how it knows? Because it sets the flags in line number 207 and 214

Reply:
initially it will have all the flags, dimension or measure, but we do reset and again find the corresponding dimension or measure and set it back, so im taking the flags before reset, to make the desicion at later part

Reply:
yes, you are correct, handled